staging/src/k8s.io/apiserver/pkg/endpoints/handlers/proxy.go

```diff
@@ -224,6 +218,8 @@ func (r *ProxyHandler) tryUpgrade(w http.ResponseWriter, req, newReq *http.Reque
 	if !httpstream.IsUpgradeRequest(req) {
 		return false
 	}
+	net.PrepareForwardedFor(newReq)
```

Comment:
shouldn't this happen when `newReq` is created above in `ServeHTTP`? Here it's only the upgrade code path, isn't it? If `httputil.NewSingleHostReverseProxy` does this already internally, please add comment here and/or before `httputil.NewSingleHostReverseProxy`.